middleware/authentication_test.go

```diff
@@ -0,0 +1,139 @@
+package middleware_test
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/lvl484/user-manager/middleware"
+	"github.com/lvl484/user-manager/mock"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/lvl484/user-manager/logger"
+	model "github.com/lvl484/user-manager/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMain(m *testing.M) {
+	logger.SetLogger(&logger.LogConfig{Output: "Stdout", Level: "debug"})
+
+	code := m.Run()
+
+	os.Exit(code)
+}
+
+func TestBasicAuthenticationMiddlewareValidPass(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mock := mock.NewMockUserProvider(ctrl)
+
+	mock.EXPECT().GetInfo("i3odja").Return(returnedValue, nil)
+
+	ba := middleware.NewBasicAuthentication(mock)
+
+	r, err := http.NewRequest("GET", "/summer", nil)
+	require.NoError(t, err)
+
+	r.SetBasicAuth("i3odja", "1q2w3e4r")
+
+	w := httptest.NewRecorder()
+
+	ba.Middleware(OK).ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusOK, w.Code)
+}
+
+func TestBasicAuthenticationMiddlewareInvalidPass(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mock := mock.NewMockUserProvider(ctrl)
+
+	mock.EXPECT().GetInfo("i3odja").Return(returnedValue, nil)
+
+	ba := middleware.NewBasicAuthentication(mock)
+
+	r, err := http.NewRequest("GET", "/summer", nil)
+	require.NoError(t, err)
+
+	r.SetBasicAuth("i3odja", "123123")
+
+	w := httptest.NewRecorder()
+
+	ba.Middleware(OK).ServeHTTP(w, r)
+
+	checkErrorResponse(t, w, http.StatusUnauthorized)
+}
+
+func TestBasicAuthenticationMiddlewareError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mock := mock.NewMockUserProvider(ctrl)
+
+	mock.EXPECT().GetInfo("i3odja").Return(returnedValue, errors.New("middleware error"))
+
+	ba := middleware.NewBasicAuthentication(mock)
+
+	r, err := http.NewRequest("GET", "/summer", nil)
+	require.NoError(t, err)
+
+	r.SetBasicAuth("i3odja", "123123")
+
+	w := httptest.NewRecorder()
+
+	ba.Middleware(OK).ServeHTTP(w, r)
+
+	checkErrorResponse(t, w, http.StatusInternalServerError)
+}
+
+func TestBasicAuthenticationMiddlewareResponseInvalid(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mock := mock.NewMockUserProvider(ctrl)
+
+	ba := middleware.NewBasicAuthentication(mock)
+
+	r, err := http.NewRequest("GET", "/summer", nil)
+	require.NoError(t, err)
+
+	w := httptest.NewRecorder()
+
+	ba.Middleware(OK).ServeHTTP(w, r)
+
+	checkErrorResponse(t, w, http.StatusUnauthorized)
```

Comment:
Please check response with expected message, not only *status code* and *body is not empty* 
`expected, err := ioutil.ReadFile("./testdata/response_unauthorized.json")`

Reply:
Good advice. I fixed it